piq/functional/base.py

```diff
@@ -54,3 +54,28 @@ def gradient_map(x: torch.Tensor, kernels: torch.Tensor) -> torch.Tensor:
     grads = torch.nn.functional.conv2d(x, kernels.to(x), padding=padding)
 
     return torch.sqrt(torch.sum(grads ** 2, dim=-3, keepdim=True))
+
+
+def pow_for_complex(base: torch.Tensor, exp: Union[int, float]) -> torch.Tensor:
+    r""" Power a 4D tensor with negative values or 5D tensor with complex values.
+    Complex numbers are represented by modulus and argument: r * \exp(i * \phi).
+
+    It will likely to be redundant with introduction of torch.ComplexTensor.
+
+    Args:
+        base: Tensor with shape (B x C x H x W) or (2 x B x C x H x W)
```

Comment:
May you meant `(B x C x H x W x 2)` is case of a complex (5D) tensor?

Reply:
No, the tensors have (2 B C H W) shapes in current implementation.

Reply:
I tried to change the implementation to `(B x C x H x W x 2)`. It leads to reduction in precision and `nan` in gradients.

Reply:
Then it is a bit weird shape, especially considering that:
1.  For SSIM/MS-SSIM implementations we have complex channels last https://github.com/photosynthesis-team/piq/blob/d3a0540fa2ea80724db11c2b5ecca08a80a7883e/piq/ssim.py#L25
2. It is kind of intuitive to have them last because we go from top to bottom: batches include channels, which consist of height and width pixels, which may be complex i.e. have two values: for real and imaginary parts respectively

@denproc would you consider re-implementing it with complex channels being last?